File: Array/Easy/1512. Number of Good Pairs.go
package main

import "fmt"

func main() {

	arr := []int{1, 1, 1, 1}
	//noOfGoodPairs := numIdenticalPairs2(arr)
	noOfGoodPairs := numIdenticalPairs2(arr)
	fmt.Println(noOfGoodPairs)
}

func numIdenticalPairs(arr []int) int {
	ans := 0

	for i := 0; i < len(arr); i++ {
		for j := i + 1; j < len(arr); j++ {
			if arr[i] == arr[j] && i < j {
				ans += 1
			}
		}
	}

	return ans
}

func numIdenticalPairs2(arr []int) int {
	ans := 0
	nums := make(map[int]int)

	for _, val := range arr {
		nums[val] += 1
	}

	for _, val := range nums {
		ans += val * (val - 1) / 2
	}

	return ans
}

File: Miscellaneous/Python/1512. Number of Good Pairs.py
class Solution:
    def numIdenticalPairs(self, nums):
        result = 0

        for i in range(len(nums)):
            for j in range(len(nums)):
                if i < j and nums[i] == nums[j]:
                    result += 1
        
        return result

obj = Solution()
nums = [1,2,3,1,1,3]
ans = obj.numIdenticalPairs(nums)
print(ans)